src/term/net_to_term.rs

```diff
@@ -328,10 +327,12 @@ pub fn net_to_term_linear(net: &INet, book: &DefinitionBook) -> (Term, bool) {
         _ => unreachable!(),
       },
       Ref { def_id } => {
-        if book.is_generated_rule(def_id) {
-          let rule = book.defs.get(&def_id).unwrap();
-
-          let mut term = rule.body.clone();
+        if book.is_generated_def(def_id) {
+          let def = book.defs.get(&def_id).unwrap();
+          if def.rules.len() > 0 {
```

Comment:
Either this is unreachable and we know the reason why, or we should be doing something here.